haproxy_discovery.sh

```diff
@@ -14,6 +14,10 @@
 HAPROXY_SOCK="/var/run/haproxy/info.sock"
 [ -n "$1" ] && echo $1 | grep -q ^/ && HAPROXY_SOCK="$(echo $1 | tr -d '\040\011\012\015')"
 
+# Define variables and perform checks
+CONFIG_FILE="$(dirname $(readlink -f "$0"))/haproxy_config"
+. $CONFIG_FILE
+
```

Comment:
All vars must be defined before this lines. Config file must be places at the same dir with scripts (Btw, I prefer `/etc/zabbix/zabbix_agentd.d/bin`)